src/Symfony/Component/Ldap/LdapClient.php

```diff
@@ -54,6 +54,8 @@ public function __construct($host = null, $port = 389, $version = 3, $useSsl = f
         $this->useSsl = (bool) $useSsl;
         $this->useStartTls = (bool) $useStartTls;
         $this->optReferrals = (bool) $optReferrals;
+
+        $this->connect();
```

Comment:
@ChadSikorra PR #17560 adresses this issue of the constructor.

@L0rD59, the connection should not be done in the constructor. Indeed, services which depend on the Ldap service may actually want to do checks before even trying to connect to the Ldap server. Like checking whether the user's password is empty or not before trying to bind against the server.